3party/CMakeLists.txt

```diff
@@ -5,6 +5,15 @@ set(CMAKE_WARN_DEPRECATED OFF CACHE BOOL "" FORCE)
 if (WITH_SYSTEM_PROVIDED_3PARTY)
   set(GFLAGS_USE_TARGET_NAMESPACE ON)
   find_package(gflags REQUIRED GLOBAL)
```

Comment:
1. We can handle it differently. On Android an older cmake is used. And usually, nobody defines WITH_SYSTEM_PROVIDED_3PARTY for android.
Then we can check the cmake version here and fail if it's not supported.

2. If WITH_SYSTEM_PROVIDED_3PARTY is the only way to build for windows, then we should enforce that.

3. It would be great to move cmake-related changes to a different PR (or even separate PRs).